src/containers/MyNdla/Learningpath/components/LearningpathStepForm.tsx

```diff
@@ -45,27 +58,117 @@ const ContentForm = styled("form", {
 
 const RADIO_GROUP_OPTIONS = ["text", "resource", "external", "folder"] as const;
 
-interface Props {
-  step?: GQLMyNdlaLearningpathStepFragment;
-  stepType: FormValues["type"];
-  onClose?: VoidFunction;
-  onDelete?: (close: VoidFunction) => Promise<void>;
-  onSave: (data: FormValues) => Promise<void>;
-}
+export const LearningpathStepForm = () => {
+  const context = useLearningpathStep();
+  const stepType = getFormTypeFromStep(context?.step);
+  const { learningpathId: learningpathIdParam, stepId: stepIdParam } = useParams();
+  const { t, i18n } = useTranslation();
 
-export const LearningpathStepForm = ({ step, stepType, onClose, onSave, onDelete }: Props) => {
-  const { t } = useTranslation();
+  const toast = useToast();
+  const navigate = useNavigate();
+  const [updateStep] = useUpdateLearningpathStep();
+  const [deleteStep] = useDeleteLearningpathStep();
+  const [createStep] = useCreateLearningpathStep();
 
-  const methods = useFormContext<FormValues>();
-  const { handleSubmit, control, reset, formState } = methods;
+  const methods = useForm<FormValues>({
+    mode: "onSubmit",
+    defaultValues: toFormValues(stepType, context?.step),
+  });
+
+  const learningpathId = learningpathIdParam ? Number(learningpathIdParam) : undefined;
 
   useEffect(() => {
-    reset(toFormValues(stepType, step));
-  }, [reset, step, stepType]);
+    if (stepIdParam === "new") setTimeout(() => document.querySelector("form")?.querySelector("input")?.focus(), 0);
+  }, [stepIdParam]);
```

Comment:
Mistenker at dette ikke funker. Er det noe vi vil ha? Burde vi isåfall bruke `autoFocus`-propen istedenfor.